app/models/mixins/compliance_mixin.rb

```diff
@@ -34,9 +36,15 @@ def scan_and_check_compliance_queue
     Compliance.scan_and_check_compliance_queue(self)
   end
 
+  # a few of these need to override
+  # name is sometimes {target_class.gsub(/_/,'')}_compliance_check
   def compliance_policies
     target_class = self.class.base_model.name.downcase
     _, plist = MiqPolicy.get_policies_for_target(self, "compliance", "#{target_class}_compliance_check")
```

Comment:
If we pull `"#{target_class}_compliance_check"` out into a method it looks like rather than overriding this "whole" method (it isn't long but still) you could just override `def compliance_check_policy_name`